sql/log_event.h

```diff
@@ -2898,14 +2898,34 @@ class Rows_log_event : public virtual binary_log::Rows_event, public Log_event {
     Key_compare(KEY **ki = nullptr) : m_key_info(ki) {}
     bool operator()(uchar *k1, uchar *k2) const {
       return key_cmp2((*m_key_info)->key_part, k1, (*m_key_info)->key_length,
-                      k2, (*m_key_info)->key_length) < 0;
+                      k2, (*m_key_info)->key_length) == 0;
     }
 
    private:
     KEY **m_key_info;
   };
-  std::set<uchar *, Key_compare> m_distinct_keys;
-  std::set<uchar *, Key_compare>::iterator m_itr;
+    class Key_hash {
+    public:
+      Key_hash(KEY **ki = nullptr) : m_key_info(ki) {}
+      size_t operator()(uchar* ptr) const {
```

Comment:
May be just create a `std::string_view` from `ptr` and `(*m_key_info)->key_length` and calculate a hash of this string_view using standard hash function.
```c++
std::string_view sv{ptr, (*m_key_info)->key_length};
return std::hash(sv);
```

Reply:
Thanks for the suggestion. Done.